framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -121,6 +122,38 @@ PenetrationLocator::detectPenetration()
 
   Threads::parallel_reduce(slave_node_range, pt);
 
+  std::vector<dof_id_type> recheck_slave_nodes = pt._recheck_slave_nodes;
+
+  if (recheck_slave_nodes.size() > 0 && _automatic_patch_update)
```

Comment:
If `_automatic_patch_update` is not turned on and there are some nodes in `recheck_slave_nodes`, maybe we should generate a warning. The downside is that users will get a lot of those messages in cases where there really are slave nodes that are not in contact with any face. We actually used to do that several years ago, but removed it because it was annoying. Now, however, we can actually do something about it, and I think that uses should really always use the option to recheck the nodes.

Reply:
I have added this warning. I will add a test for this warning.